api/v3/Profile.php

```diff
@@ -604,6 +604,7 @@ function _civicrm_api3_buildprofile_submitfields($profileID, $optionsBehaviour,
        */
     }
   }
+  $profileFields[$profileID] = is_array($profileFields[$profileID]) ? $profileFields[$profileID] : [];
```

Comment:
At this point in the test, $profileFields is an empty array, so I would expect `is_array($profileFields[$profileID])` to give an undefined index. In fact if you put an `echo 'hi';` right after this line it doesn't output, so it's giving an E_NOTICE which is inadvertently fixing it by fullfilling the fail. If you put a `var_dump($apiResult);` in assertAPIFailure itself you can see this is what's happening.

Further, if there's no valid reason for this situation, would throwing an exception directly here be a better thing to do, because always converting to a valid thing might potentially hide other issues?

Reply:
@demeritcowboy the problem is that it is not an empty array on php8 it is NULL for `$profileFields[$profileID]`

Reply:
$profileFields is an empty array, so $profileFields[$profileID] gives an ENOTICE when you use is_array on it.

Reply:
right sorry got confused, does my update work for you @demeritcowboy

Reply:
Thanks. I'm going to put merge-ready because I'm confused about a couple things, but if someone wants to merge I won't object.

1. It seems like a (unintended?) change from https://github.com/civicrm/civicrm-core/pull/21099. Before it used to say "'1000' is not a valid option for field uf_group_id", and now, with this patch, it says "Invalid value for profile_id", which in this case yes is more accurate.
2. I'm still wondering about my comment from before: If there's never a valid reason for `$profileFields[$profileID]` to be  null here, then converting it to a valid value instead of throwing an exception might end up hiding something else. In either case it's a change from before, because what it used to do before, funnily enough, was to actually create an entry for 1000 inside the array, i.e. after the uasort `$profileFields = [1000 => NULL]`, and now after the uasort `$profileFields = [1000 => []]`.

Reply:
hmm - I think - yeah good points - but I guess if we are going to do a better fix it would be fine to do it in master & if no-one actually cares enough to do it then I guess that's fine if there is no reproduceable regression.

So I'm happy to merge this & maybe someone will fix more better in master